### PR TITLE
Rest API versioning ADR

### PR DESCRIPTION
**Pull request by CDimonaco**
As the title suggest.

### CODE REVIEW

**Review by nelsonkopliku — APPROVED** [review]
Good stuff!

File: adr/0008-rest-api-versioning-strategy.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+This means that in our projects, we need to namespace the handlers, requests and responses of our http routes.
+For example in an elixir project, we want to namespace the controllers, the views and the OpenApi schemas with the version, using a module called VX (V1, V2 etc..)
+
+This means that a consumer could choose also the right Schemas for the OpenApi spec consuming, generating a client according to a specific version.
+
+The chosen approach, of URL-based API versioning, comes after a review of the most used approaches, including HTTP header-based versioning and mime-type versioning.
+
+You can read [here](https://elixirforum.com/t/how-do-you-handle-api-versioning/18898/35) and [here](https://www.troyhunt.com/your-api-versioning-is-wrong-which-is/) to have a more comprehensive overview of what we are talking about.
+
+We will have a "special" route, without any prefixed, only `/api`, this route will perform a temporary redirect to the latest API version of our services.
```
Comment (arbulu89) [inline adr/0008-rest-api-versioning-strategy.md:31]: `without any prefix` maybe.
Besides that, the sentence asks to dot after the `only /api`, thing. 
```
onlye `/api`. This route...
```

File: adr/0008-rest-api-versioning-strategy.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+We want a clear API versioning strategy for the API consumer and also a way to express unambiguously the "latest" version of our API.
+
+## Decision
+
+We will implement a URL-based API versioning.
+That means that our routes will be prefixed with `/vX`, where "X" is a progressive number that indicates the current version of the consumed API.
+This means that in our projects, we need to namespace the handlers, requests and responses of our http routes.
+For example in an elixir project, we want to namespace the controllers, the views and the OpenApi schemas with the version, using a module called VX (V1, V2 etc..)
+
+This means that a consumer could choose also the right Schemas for the OpenApi spec consuming, generating a client according to a specific version.
```
Comment (arbulu89) [inline adr/0008-rest-api-versioning-strategy.md:25]: I would remote this `This means` and put this sentence after the previous one splitted by a dot

File: adr/0008-rest-api-versioning-strategy.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+That topic will be covered in another part of the documentation
+
+We want a clear API versioning strategy for the API consumer and also a way to express unambiguously the "latest" version of our API.
+
+## Decision
+
+We will implement a URL-based API versioning.
+That means that our routes will be prefixed with `/vX`, where "X" is a progressive number that indicates the current version of the consumed API.
+This means that in our projects, we need to namespace the handlers, requests and responses of our http routes.
+For example in an elixir project, we want to namespace the controllers, the views and the OpenApi schemas with the version, using a module called VX (V1, V2 etc..)
```
Comment (arbulu89) [inline adr/0008-rest-api-versioning-strategy.md:23]: dot missing at the end of the sentence

File: adr/0008-rest-api-versioning-strategy.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+This means that a consumer could choose also the right Schemas for the OpenApi spec consuming, generating a client according to a specific version.
+
+The chosen approach, of URL-based API versioning, comes after a review of the most used approaches, including HTTP header-based versioning and mime-type versioning.
+
+You can read [here](https://elixirforum.com/t/how-do-you-handle-api-versioning/18898/35) and [here](https://www.troyhunt.com/your-api-versioning-is-wrong-which-is/) to have a more comprehensive overview of what we are talking about.
```
Comment (arbulu89) [inline adr/0008-rest-api-versioning-strategy.md:29]: I would remove the `35` from the url link. It moves you to place in the webpage that doesn't give much context

**Review by arbulu89 — COMMENTED** [review]
Fantastic work @CDimonaco ,
Just pointed some tips to improve the text, but the content is great

**Review by fabriziosestito — APPROVED** [review]
LGTM, :shipit: after addressing other reviewers comments

**Comment by CDimonaco** [comment]
Done!

**Pull request merged**